### PR TITLE
Update recommendations around disabling ThreadAutoFollow and update GA date to Q2

### PR DESCRIPTION
**Pull request by esethna**
1. Updated recommendations around disabling ThreadAutoFollow if the org isn't agile in their hardware.
2. Updated GA target date from Q1 > Q2 (I might have missed some locations)

@cwarnermm please help with text and formatting review, and if you can add any other locations where we may need to make similar changes. Perhaps we should also add similar notes re ThreadAutoFollow to the "Organizing Conversations with CRT" page

### CODE REVIEW

**Comment by esethna** [comment]
Note: This can be merged right away and isn't dependent on any eng work

File: source/install/self-managed-changelog.md [hunk 1]
```diff
@@ -21,6 +21,9 @@ Latest Mattermost Releases:
 - **v6.3.0, released 2022-01-16**
   - Original 6.3.0 release
 
+### Important Upgrade Notes
+ -  [Collapsed Reply Threads](https://docs.mattermost.com/messaging/organizing-conversations.html), available in beta, are known to have a negative impact on server performance. If you cannot easily scale up and tune your database, or if you are running the Mattermost application server and database server on the same machine, we recommended disabling [``ThreadAutoFollow``](https://docs.mattermost.com/configure/configuration-settings.html#automatically-follow-threads) and [``CollapsedThreads``](https://docs.mattermost.com/configure/configuration-settings.html#collapsed-reply-threads-beta) until Collapsed Reply Threads is promoted to general availability in Q2 2022. Learn more about these [performance considerations here](https://support.mattermost.com/hc/en-us/articles/4413183568276).
+
```
Comment (amyblais) [inline source/install/self-managed-changelog.md:26]:
```suggestion

IMPORTANT: If you upgrade from a release earlier than v6.2, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
```

File: source/configure/configuration-settings.rst [hunk 2]
```diff
@@ -5995,9 +5995,9 @@ System Admins can set the default appearance of Collapsed Reply Threads for thei
...
+   Collapsed Reply Threads are in beta. Before enabling this feature, ensure you are well versed with the `known issues <https://docs.mattermost.com/messaging/organizing-conversations.html#known-issues>`_, particularly relating to database resource requirements and server performance implications. If you cannot easily increase your database size we recommended waiting to enable Collapsed Reply Threads until it is `promoted to general availability in Q2 2022 <https://mattermost.com/blog/collapsed-reply-threads-ga/>`_.
 
-**Disabled**: Disable Collapsed Reply Threads functionality.
+**Disabled**: Disables Collapsed Reply Threads front-end functionality. To disable all backend operations and thread tracking for Collapsed Reply Threads, you must also disable ```ThreadAutoFollow`` <https://docs.mattermost.com/configure/configuration-settings.html#automatically-follow-threads>`_.
```
Comment (amyblais) [inline source/configure/configuration-settings.rst:6000]:
```suggestion
**Disabled**: Disables Collapsed Reply Threads front-end functionality. To disable all backend operations and thread tracking for Collapsed Reply Threads, you must also disable _``ThreadAutoFollow`` <https://docs.mattermost.com/configure/configuration-settings.html#automatically-follow-threads>`_. 
```

File: source/configure/configuration-settings.rst [hunk 3]
```diff
@@ -6041,12 +6041,14 @@ Automatically Follow Threads
...
 
 **True**: Threads a user starts, participates in, or is mentioned in are automatically followed. A new ``Threads`` table is added in the database that tracks threads and thread participants, and a ``ThreadMembership`` table tracks followed threads for each user and the read or unread state of each followed thread.
 
-**False**: Threads are not automatically followed and Collapsed Reply Threads cannot be enabled. If you plan to enable Collapsed Reply Threads for your instance, we don't recommend disabling this setting since the server-side tracking of followed threads and unread states is critical to create a smooth transition for end-users when enabling Collapsed Reply Threads in the future.  
+**False**: All backend operations for Collapsed Reply Threads are disabled and server perforamnce will not be impacted by the feature. Collapsed Reply Threads (``CollapsedThreads``) cannot be enabled if ``ThreadAutoFollow` is disabled.
```
Comment (amyblais) [inline source/configure/configuration-settings.rst:6051]:
```suggestion
**False**: All backend operations for Collapsed Reply Threads are disabled and server performance will not be impacted by the feature. Collapsed Reply Threads (``CollapsedThreads``) cannot be enabled if ``ThreadAutoFollow`` is disabled.    
```

File: source/configure/configuration-settings.rst [hunk 4]
```diff
@@ -5995,9 +5995,9 @@ System Admins can set the default appearance of Collapsed Reply Threads for thei
...
+   Collapsed Reply Threads are in beta. Before enabling this feature, ensure you are well versed with the `known issues <https://docs.mattermost.com/messaging/organizing-conversations.html#known-issues>`_, particularly relating to database resource requirements and server performance implications. If you cannot easily increase your database size we recommended waiting to enable Collapsed Reply Threads until it is `promoted to general availability in Q2 2022 <https://mattermost.com/blog/collapsed-reply-threads-ga/>`_.
 
-**Disabled**: Disable Collapsed Reply Threads functionality.
+**Disabled**: Disables Collapsed Reply Threads front-end functionality. To disable all backend operations and thread tracking for Collapsed Reply Threads, you must also disable the ```ThreadAutoFollow`` <https://docs.mattermost.com/configure/configuration-settings.html#automatically-follow-threads>`__.
```
Comment (cwarnermm) [inline source/configure/configuration-settings.rst:6000]:
```suggestion
**Disabled**: Disables Collapsed Reply Threads front-end functionality. To disable all backend operations and thread tracking for Collapsed Reply Threads, you must also disable the `ThreadAutoFollow <https://docs.mattermost.com/configure/configuration-settings.html#automatically-follow-threads>`__ configuration setting.
```

File: source/configure/configuration-settings.rst [hunk 5]
```diff
@@ -6041,20 +6041,22 @@ Automatically Follow Threads
...
 
 .. note::
 
    Enabling this configuration setting doesn’t retroactively follow threads for actions taken prior to the setting being enabled. For example, threads a user participated in prior to enabling this setting won't be automatically followed. However, if this setting is enabled, and a user adds a new comment on an old thread, they will now automatically be followed to the thread.  
 
-+-------------------------------------------------------------------------------------------------------------+
-| This feature's ``config.json`` setting is ``"ThreadAutoFollow": true`` with options ``true`` and ``false``. |
-+-------------------------------------------------------------------------------------------------------------+
++--------------------------------------------------------------------------------------------------------------+
+| This feature's ``config.json`` setting is ``"ThreadAutoFollow": false`` with options ``true`` and ``false``. |
```
[the remark this answers is not in the document]
Comment (amyblais) [reply source/configure/configuration-settings.rst:6058]: Will do 👍

**Comment by esethna** [comment]
@cwarnermm I think this can be merged after an approval. Thanks all!

**Review by cwarnermm — APPROVED** [review]

**Pull request merged**